src/pages/Login.js

```diff
@@ -43,22 +48,22 @@ class Login extends Component {
       }
     });
 
-    const { username, password } = this.state;
     let success = false;
-    if(formIsValid) {
-      let res = await api.login(username, password);
+    const { username, password } = this.state;
+    if (formIsValid) {
+      const res = await api.login(username, password);
       if (res.code !== api.code.ok) {
         this.setState({ error: res.msg });
```

Comment:
It is better to maintain a mapping between res.code and locallized error message.
The res.msg is for debug only.